journald/journald.hpp

```diff
@@ -65,12 +83,86 @@ struct Flags : public virtual flags::FlagsBase
           parsed_labels = _labels.get();
           return None();
         });
+
+    add(&Flags::max_size,
+        "max_size",
+        "Maximum size, in bytes, of a single log file.\n"
+        "Defaults to 10 MB.  Must be at least 1 (memory) page.",
+        Megabytes(10),
+        [](const Bytes& value) -> Option<Error> {
+          if (value.bytes() < os::pagesize()) {
+            return Error(
+                "Expected --max_size of at least " +
+                stringify(os::pagesize()) + " bytes");
+          }
+          return None();
+        });
+
+    add(&Flags::logrotate_options,
+        "logrotate_options",
+        "Additional config options to pass into 'logrotate'.\n"
+        "This string will be inserted into a 'logrotate' configuration file.\n"
+        "i.e.\n"
+        "  /path/to/<log_filename> {\n"
+        "    <logrotate_options>\n"
+        "    size <max_size>\n"
+        "  }\n"
+        "NOTE: The 'size' option will be overriden by this command.");
+
+    add(&Flags::log_filename,
+        "log_filename",
+        "Absolute path to the leading log file.\n"
+        "NOTE: This command will also create two files by appending\n"
+        "'" + CONF_SUFFIX + "' and '" + STATE_SUFFIX + "' to the end of\n"
+        "'--log_filename'.  These files are used by 'logrotate'.",
+        [](const Option<std::string>& value) -> Option<Error> {
+          if (value.isNone()) {
+            return Error("Missing required option --log_filename");
+          }
+
+          if (!path::absolute(value.get())) {
+            return Error("Expected --log_filename to be an absolute path");
+          }
+
+          return None();
+        });
+
+    add(&Flags::logrotate_path,
+        "logrotate_path",
+        "If specified, this command will use the specified\n"
+        "'logrotate' instead of the system's 'logrotate'.",
+        "logrotate",
+        [](const std::string& value) -> Option<Error> {
+          // Check if `logrotate` exists via the help command.
+          // TODO(josephw): Consider a more comprehensive check.
+          Try<std::string> helpCommand =
+            os::shell(value + " --help > /dev/null");
```

Comment:
Nope.  The environment is inherited from the agent (with some subtractions).  The user can't change this option.